subsys/bluetooth/rpc/host/bt_rpc_conn_host.c

```diff
@@ -0,0 +1,1433 @@
+}
+
+NRF_RPC_CBOR_CMD_DECODER(bt_rpc_grp, bt_conn_enc_key_size, BT_CONN_ENC_KEY_SIZE_RPC_CMD,
+			 bt_conn_enc_key_size_rpc_handler, NULL);
+#endif /* defined(CONFIG_BT_SMP) */
+
+static void bt_conn_cb_connected_call(struct bt_conn *conn, uint8_t err)
+{
+	struct nrf_rpc_cbor_ctx _ctx;
+	size_t _buffer_size_max = 5;
+
+	NRF_RPC_CBOR_ALLOC(_ctx, _buffer_size_max);
+
+	bt_rpc_encode_bt_conn(&_ctx.encoder, conn);
+	ser_encode_uint(&_ctx.encoder, err);
+
+	nrf_rpc_cbor_cmd_no_err(&bt_rpc_grp, BT_CONN_CB_CONNECTED_CALL_RPC_CMD,
+				&_ctx, ser_rsp_decode_void, NULL);
+}
+
+static void bt_conn_cb_disconnected_call(struct bt_conn *conn, uint8_t reason)
+{
+	struct nrf_rpc_cbor_ctx _ctx;
+	size_t _buffer_size_max = 5;
+
+	NRF_RPC_CBOR_ALLOC(_ctx, _buffer_size_max);
+
+	bt_rpc_encode_bt_conn(&_ctx.encoder, conn);
+	ser_encode_uint(&_ctx.encoder, reason);
+
+	nrf_rpc_cbor_cmd_no_err(&bt_rpc_grp, BT_CONN_CB_DISCONNECTED_CALL_RPC_CMD,
+				&_ctx, ser_rsp_decode_void, NULL);
+}
+
+static bool bt_conn_cb_le_param_req_call(struct bt_conn *conn, struct bt_le_conn_param *param)
+{
+	struct nrf_rpc_cbor_ctx _ctx;
+	bool _result;
+	size_t _buffer_size_max = 15;
+
+	NRF_RPC_CBOR_ALLOC(_ctx, _buffer_size_max);
+
+	bt_rpc_encode_bt_conn(&_ctx.encoder, conn);
+	bt_le_conn_param_enc(&_ctx.encoder, param);
+
+	nrf_rpc_cbor_cmd_no_err(&bt_rpc_grp, BT_CONN_CB_LE_PARAM_REQ_CALL_RPC_CMD,
+				&_ctx, ser_rsp_decode_bool, &_result);
+
+	return _result;
+}
+
+static void bt_conn_cb_le_param_updated_call(struct bt_conn *conn, uint16_t interval, uint16_t
+					     latency, uint16_t timeout)
+{
+	struct nrf_rpc_cbor_ctx _ctx;
+	size_t _buffer_size_max = 12;
+
+	NRF_RPC_CBOR_ALLOC(_ctx, _buffer_size_max);
+
+	bt_rpc_encode_bt_conn(&_ctx.encoder, conn);
+	ser_encode_uint(&_ctx.encoder, interval);
+	ser_encode_uint(&_ctx.encoder, latency);
+	ser_encode_uint(&_ctx.encoder, timeout);
+
+	nrf_rpc_cbor_cmd_no_err(&bt_rpc_grp, BT_CONN_CB_LE_PARAM_UPDATED_CALL_RPC_CMD,
+				&_ctx, ser_rsp_decode_void, NULL);
+}
+
+#if defined(CONFIG_BT_SMP)
+static void bt_conn_cb_identity_resolved_call(struct bt_conn *conn, const bt_addr_le_t *rpa, const
+					      bt_addr_le_t *identity)
+{
+	struct nrf_rpc_cbor_ctx _ctx;
+	size_t _buffer_size_max = 9;
+
+	_buffer_size_max += rpa ? sizeof(bt_addr_le_t) : 0;
+	_buffer_size_max += identity ? sizeof(bt_addr_le_t) : 0;
+
+	NRF_RPC_CBOR_ALLOC(_ctx, _buffer_size_max);
+
+	bt_rpc_encode_bt_conn(&_ctx.encoder, conn);
+	ser_encode_buffer(&_ctx.encoder, rpa, sizeof(bt_addr_le_t));
+	ser_encode_buffer(&_ctx.encoder, identity, sizeof(bt_addr_le_t));
+
+	nrf_rpc_cbor_cmd_no_err(&bt_rpc_grp, BT_CONN_CB_IDENTITY_RESOLVED_CALL_RPC_CMD,
+				&_ctx, ser_rsp_decode_void, NULL);
+}
+
+static void bt_conn_cb_security_changed_call(struct bt_conn *conn, bt_security_t level,
+					     enum bt_security_err err)
+{
+	struct nrf_rpc_cbor_ctx _ctx;
+	size_t _buffer_size_max = 13;
+
+	NRF_RPC_CBOR_ALLOC(_ctx, _buffer_size_max);
```

Comment:
This function is also misinforming about what it does. The data is declared on the stack, and the "free" function does nothing.
If this is supposed to be made dynamic illusion is not complete because there isn't any "handling" of possible failure to allocate here. So now I need to understand what the macro is doing in order for me to see that I don't have to handle that.

Reply:
`NRF_RPC_CBOR_ALLOC` macro is part of nRF PRC and it is already in, so we have to create a new PR to change it.

It allocates the memory, but where is does, it depends on nRF PRC backend. Currently we have only RPMsg backend, that allocates memory on the stack, but different backends may allocate the memory in different places, e.g. I have draft of  nRF RPC backend that works directly no shared memory and IPC peripheral and `NRF_RPC_CBOR_ALLOC` allocates memory dynamically directly on the shared memory area. I can imagine that UART backend can also allocate memory directly on some io buffer instead on stack to avoid copying.

Additionally, reserving some memory on the stack is sometimes also called allocating, e.g. `alloca` from `stdlib`.